src/reducers/select.js

```diff
@@ -47,65 +48,91 @@ export default (pattern) => (base, source) => {
     }
     // Now that there is only one field in both baseFields and sourceFields, the arrays can be destructured into variables
     const [baseField] = baseFields;
-    // Debug(`baseField destructured: ${JSON.stringify(baseField, undefined, 2)}`);
     const [sourceField] = sourceFields;
-    // Debug(`sourceField destructured: ${JSON.stringify(sourceField, undefined, 2)}`);
 
     // Check that the base and source tags are equal
-    // Test 03: Base and source tags are equal for one field (continue to comparison of contents)
+    // Test 03: Base and source tags are equal for one field (continue to normalization and comparison of contents)
     // Test 04: Base and source tags are not equal (base field returned unmodified)
     // Note: to check this, the pattern has to allow two tags, otherwise the unequal tag does not even pass source.get(pattern)
     if (baseField.tag === sourceField.tag === false) {
       debug(`Tags are not equal`);
       return base;
     }
     const baseSubs = baseField.subfields;
+    debug(`baseSubs: ${JSON.stringify(baseSubs, undefined, 2)}`);
     const sourceSubs = sourceField.subfields;
-
-    // Test 05: Normalize subfield values
-    baseSubs.forEach(normalizeSubfieldValue);
-    debug(`baseField normalized: ${JSON.stringify(baseField, undefined, 2)}`);
-    sourceSubs.forEach(normalizeSubfieldValue);
-    debug(`sourceField normalized: ${JSON.stringify(sourceField, undefined, 2)}`);
+    debug(`sourceSubs: ${JSON.stringify(baseSubs, undefined, 2)}`);
+
+    // Test 05: Normalize subfield values (base and source are equal) --> return base
+    // Use clonedeep to make independent copies of the baseSubs and sourceSubs arrays
+    // These copies can be modified by normalization without affecting the original base and source
+    // Mapping baseSubsNormalized = baseSubs.map(normalizeSubfieldValue) does not work, it modifies the originals
+    const baseSubsNormalized = clonedeep(baseSubs);
+    baseSubsNormalized.forEach(normalizeSubfieldValue);
```

Comment:
Unnecessary, just use `.map`, destructure `code` and `value`, and return a new object which has the same code and a new value for `value`, for example:
```js
.map(({code, value}) => {
  return {code, value: doSomething()};

  function doSomething() {
    return value.replace(/foo/, 'bar');
  }
});